### PR TITLE
feat: [CDS-70207]: Add interfaces and implement helm chart fetch logic

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
Adding implementation for checking if manifest does support a delegate task and each manifest handler can create their own task. 
This PR defines the next flow:
- First manifest is checked if does support delegate task (ManifestTaskHandler#isSupported)
- Then delegate task is created (ManifestTaskHandler#createTaskData), here only the task data is created, queueing part will be handled independently as is expected to not be manifest dependent
- Once the task response is received then  ManifestTaskHandler#updateManifestOutcome. Handler is responsible of generating a new manifest outcome based on the task response. Ideally manifest outcome should be immutable so a new instance should be created containing previous and updated state. But if this is not possible is acceptable to update existing one
- If no changes are required to be passed to manifest we can return Optional.empty(), this will result in not updating manifest content

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49960)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger uts

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger uts

**Comment by abosii-harness** [comment]
trigger ut0, ut3

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/task/HelmChartManifestTaskHandler.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import com.google.inject.Inject;
+import java.time.Duration;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@OwnedBy(HarnessTeam.CDP)
+public class HelmChartManifestTaskHandler implements ManifestTaskHandler {
+  public static final long DEFAULT_FETCH_TIMEOUT_MILLIS = Duration.ofMinutes(5).toMillis();
```
Comment (acasian-harness) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/task/HelmChartManifestTaskHandler.java:42]: 10?

**Review by acasian-harness — APPROVED** [review]
Easy to review, great PR!

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/task/HelmChartManifestTaskHandler.java (hunk 1 above)
Comment (abosii-harness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/task/HelmChartManifestTaskHandler.java:42]: yeah, make sense

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger review

**Pull request merged**